octproz/src/glwindow2d.cpp

```diff
@@ -288,24 +289,48 @@ void GLWindow2D::paintGL(){
-		float ratioTexture = textureWidth/textureHeight;
-		float ratioScreen = screenWidth/screenHeight;
-		if(ratioTexture > ratioScreen){
-			this->screenHeightScaled = (screenWidth/ratioTexture)/screenHeight;
-		}
-		else{
-			this->screenWidthScaled = (screenHeight*ratioTexture)/screenWidth;
-		}
-	}
-	//recalculate marker coordinates
-	this->slot_setMarkerPosition(this->markerPosition);
+    double deg_to_rad = 3.1415926535898/180.0;
+    double rotationAngle_rad = rotationAngle*deg_to_rad;
+
+    if(this->keepAspectRatio){
+        float textureWidth = static_cast<float>(this->height);
+        float textureHeight = static_cast<float>(this->width);
+        float screenWidth = static_cast<float>(w);//(float)this->size().width();
+        float screenHeight = static_cast<float>(h);//(float)this->size().height();
+        float ratioTexture = textureWidth/textureHeight;
+        float ratioScreen = screenWidth/screenHeight;
+        this->screenWidthScaled = 1;
+        this->screenHeightScaled = 1;
+
+        if(ratioTexture > ratioScreen){
+            this->screenHeightScaled = ratioScreen/ratioTexture;
+        }
+        else{
+            this->screenWidthScaled =  ratioTexture/ratioScreen;           
+        }
+        this->screenWidthScaled = this->screenWidthScaled*screenWidth/screenHeight; //transformation to uniformly x-axis
+        //vertex01 bottom left
+        this->screenWidthScaled01 = -this->screenWidthScaled*cos(rotationAngle_rad)+this->screenHeightScaled*sin(rotationAngle_rad); //rotaion x-coordinate
+        this->screenHeightScaled01 = -this->screenWidthScaled*sin(rotationAngle_rad)-this->screenHeightScaled*cos(rotationAngle_rad); //rotation y-coordinate
+        this->screenWidthScaled01 = screenWidthScaled01*screenHeight/screenWidth; //back-transformation to streched x-axis
+        //vertex11 top left
+        this->screenWidthScaled11 = -this->screenWidthScaled*cos(rotationAngle_rad)-this->screenHeightScaled*sin(rotationAngle_rad);
+        this->screenHeightScaled11 = -this->screenWidthScaled*sin(rotationAngle_rad)+this->screenHeightScaled*cos(rotationAngle_rad);
+        this->screenWidthScaled11 = screenWidthScaled11*screenHeight/screenWidth;
+        //vertex10 top right
+        this->screenWidthScaled10 = this->screenWidthScaled*cos(rotationAngle_rad)-this->screenHeightScaled*sin(rotationAngle_rad);
+        this->screenHeightScaled10 = this->screenWidthScaled*sin(rotationAngle_rad)+this->screenHeightScaled*cos(rotationAngle_rad);
+        this->screenWidthScaled10 = screenWidthScaled10*screenHeight/screenWidth;
+        //vertex00 botton right
+        this->screenWidthScaled00 = this->screenWidthScaled*cos(rotationAngle_rad)+this->screenHeightScaled*sin(rotationAngle_rad);
+        this->screenHeightScaled00 = this->screenWidthScaled*sin(rotationAngle_rad)-this->screenHeightScaled*cos(rotationAngle_rad);
+        this->screenWidthScaled00 = screenWidthScaled00*screenHeight/screenWidth;
+    }

```

Comment:
suggestion: some calculations are done multiple times (e.g. this->screenHeightScaled*sin(rotationAngle_rad)). Maybe each calculation can be done just once and the corresponding results could be used to calculate the rotated vertex coordinates (screenHeightScaled01, screenHeightScaled11, ...) ?